test/test_wave.py

```diff
@@ -98,165 +98,104 @@ def _cos(self, val):
 
     for n in [4, 8, 16]:
-        from meshmode.mesh.generation import generate_regular_rect_mesh
-        mesh = generate_regular_rect_mesh(
-                a=(-1.,)*dim,
-                b=(1.,)*dim,
-                n=(n,)*dim)
+        mesh = mesh_factory(n)
 
         from grudge.eager import EagerDGDiscretization
         discr = EagerDGDiscretization(cl_ctx, mesh, order=order)
 
         nodes = discr.nodes().with_queue(queue)
 
-        # 2D: phi(x,y,t) = cos(omega*t-pi/4)*(x-1)^3*(x+1)^3*(y-1)^3*(y+1)^3
-        # 3D: phi(x,y,z,t) = cos(omega*t-pi/4)*(x-1)^3*(x+1)^3*(y-1)^3*(y+1)^3
-        #                      *(z-1)^3*(z+1)^3
-        # Not actual solutions, but we can still apply wave operator to them
-        coord_names = ["x", "y", "z"]
-        sym_coords = [pmbl.var(name) for name in coord_names]
+        sym_coords = [pmbl.var(name) for name in ["x", "y", "z"]]
         sym_t = pmbl.var("t")
-        sym_c = pmbl.var("c")
-        sym_omega = pmbl.var("omega")
-        sym_pi = pmbl.var("pi")
-        sym_cos = pmbl.var("cos")
-        sym_phi = sym_cos(sym_omega*sym_t - sym_pi/4)
-        for i in range(dim):
-            sym_phi = sym_phi * (sym_coords[i]-1)**3 * (sym_coords[i]+1)**3
+
+        # f = phi_tt - c^2 * div(grad(phi))
+        sym_f = sym_diff(sym_t)(sym_diff(sym_t)(sym_phi)) - c**2\
+                    * sym_div(sym_grad(dim, sym_phi))
 
         # u = phi_t
         sym_u = sym_diff(sym_t)(sym_phi)
 
         # v = c*grad(phi)
-        sym_v = [sym_c * sym_diff(sym_coords[i])(sym_phi) for i in range(dim)]
+        sym_v = [c * sym_diff(sym_coords[i])(sym_phi) for i in range(dim)]
 
-        # rhs(u part) = c*div(v)
+        # rhs(u part) = c*div(v) + f
         # rhs(v part) = c*grad(u)
         sym_rhs = flat_obj_array(
-                    sym_c * sym_div(sym_v),
-                    make_obj_array([sym_c]) * sym_grad(dim, sym_u))
-
-        c = 2.
-
-        eval_values = {
-            "t": 0.,
-            "c": c,
-            "omega": 1.,
-            "pi": np.pi
-        }
-        eval_values.update({coord_names[i]: nodes[i] for i in range(dim)})
+                    c * sym_div(sym_v) + sym_f,
+                    make_obj_array([c]) * sym_grad(dim, sym_u))
 
-        def sym_eval(expr):
+        def sym_eval(expr, t):
+            coord_names = ["x", "y", "z"]
+            eval_values = {coord_names[i]: nodes[i] for i in range(dim)}
+            eval_values["t"] = t
             return EvaluationMapper(eval_values)(expr)
 
-        u = sym_eval(sym_u)
-        v = sym_eval(sym_v)
-        fields = flat_obj_array(u, v)
+        u = sym_eval(sym_u, 0.)
+        v = sym_eval(sym_v, 0.)
+        f = sym_eval(sym_f, 0.)
 
         from mirgecom.wave import wave_operator
-        rhs = wave_operator(discr, c=c, w=fields)
+        rhs = wave_operator(discr, c=c, w=flat_obj_array(u, v))
+        rhs[0] = rhs[0] + f
 
-        expected_rhs = sym_eval(sym_rhs)
+        expected_rhs = sym_eval(sym_rhs, 0.)
```

Comment:
Try another time point. `t=0` is potentially special.

Reply:
Yeah, this will be going away shortly. This PR is mainly a stepping stone towards adding time integration to the tests (wanted to combine first so I have less code to change).